lib/node/remix.ts

```diff
@@ -1,22 +1,40 @@
 import { defineConventionalRoutes } from "@remix-run/dev/config/routesConvention";
-import type { ConfigRoute, RouteManifest } from "@remix-run/dev/config/routes";
+import {
+  ConfigRoute,
+  RouteManifest,
+  defineRoutes,
+} from "@remix-run/dev/config/routes";
+import type { PluginOptions } from "./index";
+import type { AppConfig } from "@remix-run/dev/config";
+import type { RequireOnly } from "./utils";
 
-interface Options {
-  appDir: string;
-  is404Route?: (route: Route) => boolean;
-}
+export type RemixOptions = Pick<
+  AppConfig,
+  "appDirectory" | "routes" | "ignoredRouteFiles"
+>;
```

Comment:
The nice thing about this is if you decide to support more remix options it's easy to add.